File: pycode/leetcode415.py
class leetcode415:
    def addStrings(self, num1: str, num2: str) -> str:
        i, j = len(num1) - 1, len(num2) - 1
        carry = 0
        ans = []

        while i >= 0 or j >= 0 or carry:
            x = int(num1[i]) if i >= 0 else 0
            y = int(num2[j]) if j >= 0 else 0
            total = x + y + carry
            ans.append(str(total % 10))
            carry = total // 10
            i -= 1
            j -= 1

        return ''.join(reversed(ans))

if __name__ == "__main__":
    solution = leetcode415()
    print(solution.addStrings("11", "123"))
    print(solution.addStrings("456", "77"))
    print(solution.addStrings("0", "0"))

File: jsCode/leetcode415.js
class leetcode415 {
    addStrings(num1, num2) {
        let i = num1.length - 1;
        let j = num2.length - 1;
        let carry = 0;
        let result = [];

        while (i >= 0 || j >= 0 || carry !== 0) {
            let x = i >= 0 ? parseInt(num1[i]) : 0;
            let y = j >= 0 ? parseInt(num2[j]) : 0;
            let sum = x + y + carry;
            result.push(sum % 10);
            carry = Math.floor(sum / 10);
            i--;
            j--;
        }

        return result.reverse().join('');
    }
}

// 测试代码
const solution = new leetcode415();
console.log(solution.addStrings("11", "123"));
console.log(solution.addStrings("456", "77"));
console.log(solution.addStrings("0", "0"));
